src/internal/log/http.go

```diff
@@ -44,7 +46,43 @@ func AddLoggerToHTTPServer(rctx context.Context, name string, s *http.Server) {
 			ctx = ChildLogger(ctx, "", WithFields(id))
 			ctx = metadata.NewOutgoingContext(ctx, metadata.MD{"x-request-id": requestID})
 			r = r.WithContext(ctx)
+
+			var respStatusCode int
+			var errbuf []byte
+			w = httpsnoop.Wrap(w, httpsnoop.Hooks{
+				WriteHeader: func(next httpsnoop.WriteHeaderFunc) httpsnoop.WriteHeaderFunc {
```

Comment:
There appears to be `Header` which might handle the case where `WriteHeader` isn't called?  (The assumption below that the status code is 200 is correct.)

Also, now that we have a spy for headers, can we include the response headers in the log message?

Reply:
N.B.: I did try adding a wrapper for `Header`, but it doesn't seem to handle the status-code 200 case, so I left the wrapper as-is.